zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -318,6 +318,253 @@ def test_security_checklist(self):
                 expected_to_pass=expected_to_pass)
 
 
+class NeutronOpenvSwitchTest(test_utils.OpenStackBaseTest):
+    """Test basic Neutron Openvswitch Charm functionality."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running Neutron Openvswitch tests."""
+        super(NeutronOpenvSwitchTest, cls).setUpClass()
+
+        cls.current_os_release = openstack_utils.get_os_release()
+
+        cls.compute_unit = zaza.model.get_units('nova-compute')[0]
+        cls.rabbitmq_unit = zaza.model.get_units('rabbitmq-server')[0]
+        cls.neutron_api_unit = zaza.model.get_units('neutron-api')[0]
+        cls.n_ovs_unit = zaza.model.get_units('neutron-openvswitch')[0]
+
+        cls.bionic_stein = openstack_utils.get_os_release('bionic_stein')
+        cls.trusty_mitaka = openstack_utils.get_os_release('trusty_mitaka')
+
+        if cls.current_os_release >= cls.bionic_stein:
+            cls.pgrep_full = True
+        else:
+            cls.pgrep_full = False
+
+        # set up client
+        cls.neutron_client = (
+            openstack_utils.get_neutron_session_client(cls.keystone_session))
+
+    def test_100_services(self):
+        """Verify the expected services are running."""
+        logging.debug('Checking system services on units...')
+
+        services = {
+            self.compute_unit.entity_id: ['nova-compute',
+                                          'neutron-plugin-openvswitch-agent'],
+            self.rabbitmq_unit.entity_id: ['rabbitmq-server'],
+            self.neutron_api_unit.entity_id: ['neutron-server'],
+        }
+
+        if self.current_os_release >= self.trusty_mitaka:
+            services[self.compute_unit.entity_id] = [
+                'nova-compute',
+                'neutron-openvswitch-agent'
+            ]
+
+        for unit_name, unit_services in services.items():
+            logging.info('Checking services on {}'.format(unit_name))
```

Comment:
This is probably just a style thing I've gotten used to from linters

cite: 
-   https://docs.python.org/3/howto/logging.html#logging-variable-data 
-   https://stackoverflow.com/questions/29147442/how-to-fix-pylint-logging-not-lazy
```suggestion
            logging.info('Checking services on %s', unit_name)
```

Reply:
it's how the style is being used for all the zaza tests, so used the same process :)